staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1746,6 +1749,42 @@ type CSIVolumeSource struct {
 	NodePublishSecretRef *LocalObjectReference `json:"nodePublishSecretRef,omitempty" protobuf:"bytes,5,opt,name=nodePublishSecretRef"`
 }
 
+// Represents ephemeral storage that is handled by a normal storage driver.
+type EphemeralVolumeSource struct {
+	// Will be created as a stand-alone PVC to provision the volume.
+	//
+	// The pod in which this EphemeralVolumeSource is embedded will be the
+	// owner, i.e. the PVC will be deleted together with the pod.
+	// The name of the PVC will be `<pod name>-<volume name>` where `<volume name>`
+	// is the name from the `PodSpec.Volumes` array entry. Care must
+	// be taken to keep this concatendated name shorter enough that it
+	// can be used as object name.
+	//
+	// Labels and annotations will be copied from the ObjectMeta
+	// inside the template. All other meta data in it is ignored.
+	//
+	// This field is read-only and no changes will be made by Kubernetes
+	// to the PVC after it has been created.
+	//
+	// Required, must not be nil.
+	VolumeClaimTemplate *PersistentVolumeClaimTemplate `json:"volumeClaimTemplate,omitempty" protobuf:"bytes,1,opt,name=volumeClaimTemplate"`
+
+	// Specifies a read-only configuration for the volume.
+	// Defaults to false (read/write).
+	// +optional
+	ReadOnly bool `json:"readOnly,omitempty" protobuf:"varint,2,opt,name=readOnly"`
+}
+
+// PersistentVolumeClaimTemplate is used to produce
+// PersistentVolumeClaim objects.
+type PersistentVolumeClaimTemplate struct {
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
You need to describe which fields are used and ignored.

Reply:
That was under `EphemeralVolumeSource`. I agree, it makes more sense here.

Reply:
In general it's ok to be specific here, and reference or cover use case driven (duplicate / summarize / highlight specific fields) in the field apidoc.  That way you look at the field to figure out how to use the object as a whole, and can drill down to specific fields if necessary.